### ISSUE TITLE
Adopt Julia Style

### ISSUE THREAD

**Issue by klensink** [XtractOpen/Meganet.jl#10]
I think it would be a good idea to start following Julia's style conventions moving forward. We could  slowly go back and change things that are already in place until we catch up. 

I've quoted some sections of [Julia's style guide](https://docs.julialang.org/en/latest/manual/style-guide/) for things that stand out to me:

1) __Camel Casing__
All of Julia's Base (and the vast majority of all packages) follow a particular style, so it might be difficult for users to adopt the package if we use a different style.

> - modules and type names use capitalization and camel case: module SparseArrays, struct UnitRange.
> - functions are lowercase (maximum(), convert()) and, when readable, with multiple words squashed together (isequal(), haskey()). When necessary, use underscores as word separators. Underscores are also used to indicate a combination of concepts (remotecall_fetch() as a more efficient implementation of fetch(remotecall(...))) or as modifiers (sum_kbn()).
> - conciseness is valued, but avoid abbreviation (indexin() rather than indxin()) as it becomes difficult to remember whether and how particular words are abbreviated.

2) __Re-naming Outer Constructors__
`function getAffineScalingLayer(TYPE::Type, nData)
    return AffineScalingLayer{TYPE}(nData)
end
`
[Should be](https://docs.julialang.org/en/stable/manual/constructors/):

AffineScalingLayer(TYPE::type, nData) = AffineScalingLayer{TYPE}(nData)

3) __Unnecessary Static Type Parameters__

>A function signature:
>`foo(x::T) where {T<:Real} = ...`
>should be written as:
>`foo(x::Real) = ...`

or `foo(x::Array{T, 2}) where {T<:Number}` should be written as `foo(x::Array{<:Number, 2})` if `T` is not used elsewhere.

**Comment by eldadHaber** [comment]
I hate underscores!

**Comment by lruthotto** [comment]
I'd put changing the style on low priority. let's focus on performance first.

**Comment by klensink** [comment]
I agree that it is a low priority, but I just wanted to bring it up because keeping it in mind going forward will reduce the amount of changes we'll need to make later.
